### PR TITLE
REF/TST: collect reindex tests

### PR DESCRIPTION
**Pull request by jbrockmendel**
- [ ] closes #xxxx
- [ ] tests added / passed
- [ ] passes `black pandas`
- [ ] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [ ] whatsnew entry

### CODE REVIEW

File: pandas/tests/frame/test_axis_select_reindex.py [hunk 1]
```diff
@@ -36,352 +33,6 @@ def test_merge_join_different_levels(self):
             result = df1.join(df2, on="a")
         tm.assert_frame_equal(result, expected)
 
-    def test_reindex(self, float_frame):
```
Comment (jreback) [inline pandas/tests/frame/test_axis_select_reindex.py:39]: let's rename this file then

File: pandas/tests/frame/test_axis_select_reindex.py [hunk 2]
```diff
@@ -36,352 +33,6 @@ def test_merge_join_different_levels(self):
             result = df1.join(df2, on="a")
```
Comment (jreback) [inline pandas/tests/frame/test_axis_select_reindex.py:33]: what is left in this file? also some aligns left. i would rather have you simply rename this an extract the non-reindex tests (which seems to be very few)

**Review by jreback — CHANGES_REQUESTED** [review]

File: pandas/tests/frame/test_axis_select_reindex.py (hunk 2 above)
Comment (jbrockmendel) [reply pandas/tests/frame/test_axis_select_reindex.py:33]: > also some aligns left.

theres one with "align" in the name but its testing arithmetic, not the align method directly.

File: pandas/tests/frame/test_axis_select_reindex.py (hunk 1 above)
Comment (jbrockmendel) [reply pandas/tests/frame/test_axis_select_reindex.py:39]: sure, will update

**Comment by jreback** [comment]
azure didn't run here for some reason as well

**Review by jreback — APPROVED** [review]

**Comment by jreback** [comment]
cool merge on green-ish

**Comment by jbrockmendel** [comment]
failures are due to pyarrow upgrade (currently troubleshooting locally).  greenish

**Pull request merged**